src/jrd/replication/ChangeLog.cpp

```diff
@@ -950,11 +1003,26 @@ ChangeLog::Segment* ChangeLog::reuseSegment(ChangeLog::Segment* segment)
 	// If renaming fails, then we just create a new file.
 	// The old segment will be reused later in this case.
 	if (::rename(orgname.c_str(), newname.c_str()) < 0)
+	{
+#ifdef DEV_BUILD
+		const auto err = ERRNO;
+
+		string warn;
+		warn.printf("Journal file %s rename to %s failed (error %d)",
+					orgname.c_str(), newname.c_str(), err);
+
+		logPrimaryWarning(m_config->dbName, warn);
+#endif
 		return createSegment();
+	}
 
 	// Re-open the segment using a new name and initialize it
 
 	const auto fd = os_utils::openCreateSharedFile(newname.c_str(), O_BINARY);
+	if (fd == -1)
+	{
+		raiseError("Journal file %s create() failed (error %d)", newname.c_str(), errno);
+	}
 
```

Comment:
`openCreateSharedFile()` throws itself, so this addition is not necessary.

Reply:
Done